django/forms/models.py

```diff
@@ -1009,9 +1009,15 @@ def _get_foreign_key(parent_model, model, fk_name=None, can_fail=False):
         fks_to_parent = [f for f in opts.fields if f.name == fk_name]
         if len(fks_to_parent) == 1:
             fk = fks_to_parent[0]
-            if not isinstance(fk, ForeignKey) or \
-                    (fk.remote_field.model != parent_model and
-                     fk.remote_field.model not in parent_model._meta.get_parent_list()):
+            remote_field_is_proxy = fk.remote_field and fk.remote_field.model._meta.proxy
```

Comment:
In both cases we're checking `fk.remote_field.model._meta.proxy` when `fk` is a `ForeignKey` so there is no need to check `fk.remote_field`.